woocommerce/woocommerce
# Add To Cart Form: Add the `Stepper` style

> [!IMPORTANT]  
> This PR impacts a critical part of the user flow. When you review it, please consider any potential issues.
> I'm concerned that the new CSS might conflict with block themes that customize the appearance of the Add To Cart button.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #42299.

This PR improves the Add to Cart with Options block:
- it moves the block from the Atomic folder to the Blocks folder (check #42470).
- It adds the Quantity Selector option to change the block's style.
- it improves the default style.

### It moves the block from the Atomic folder to the Blocks folder 

With #42470, we want to move all the product element blocks from the atomic folder. I did in this PR because it was necessary for tweaking the webpack configuration to add the Interactivity API support to this block.


### It adds the Quantity Selector option to change the block's style


https://github.com/woocommerce/woocommerce/assets/4463174/15ea6e55-40d1-4cee-ad8a-0c12318da001

This PR adds a new option to change the block's style. Also, when the `stepper` mode is enabled, the Interactivity API changes the input value when the user clicks on the `-` or `+` button.

~~### it improves the default style~~

~~This PR improves the default style (now called `input` mode). Now, the input has the same height of the button:~~

| Before | After |
|--------|--------|
|![image](https://github.com/woocommerce/woocommerce/assets/4463174/6a85a121-a00e-42a5-b360-1aa9aef09c8d)|![image](https://github.com/woocommerce/woocommerce/assets/4463174/a2c22bb7-181b-4184-a0e1-33c7064ebe45)| 




## Screenshots

### Single Product Template

#### Input

|  Product Type  | TT4 | TT3 |
|--------|--------|--------|
| Simple Product |![image](https://github.com/woocommerce/woocommerce/assets/4463174/55ffe693-2c60-4c00-9455-21f72009c97b) |![image](https://github.com/woocommerce/woocommerce/assets/4463174/25df8fcb-41e8-450d-bbb4-b84d0db07371)|
| Product Variation |![image](https://github.com/woocommerce/woocommerce/assets/4463174/f58b0c2d-b047-47b2-ba5d-5a06ef51c921)| ![image](https://github.com/woocommerce/woocommerce/assets/4463174/a8490a17-d328-4a1a-bdf0-5005b483068d) |
| Grouped Product |![image](https://github.com/woocommerce/woocommerce/assets/4463174/436a4382-7ce6-4611-9a3b-6700bdd157b0)|![image](https://github.com/woocommerce/woocommerce/assets/4463174/52cd48f1-2f8f-4c62-b7b7-300e6c8cbe34) | 

#### Stepper

|  Product Type  | TT4 | TT3 |
|--------|--------|--------|
| Simple Product | ![image](https://github.com/woocommerce/woocommerce/assets/4463174/3edb9901-f3f0-4b15-9b58-fb5c631b4e14)| ![image](https://github.com/woocommerce/woocommerce/assets/4463174/7d50b8ef-8ce4-4cc3-8f03-e5d3e40f3dfb) |
| Product Variation | ![image](https://github.com/woocommerce/woocommerce/assets/4463174/c0f01b10-32da-4182-a105-8425c24e444e)| ![image](https://github.com/woocommerce/woocommerce/assets/4463174/4a2e75e4-628d-4bba-82ce-9c40e002ca8a) |
| Grouped Product | ![image](https://github.com/woocommerce/woocommerce/assets/4463174/c5a2a5d0-32e9-4aca-a981-8c7c9f902f08) | ![image](https://github.com/woocommerce/woocommerce/assets/4463174/46695d9a-f6df-4e83-aadd-551342acb611)| 


### Single Product Block


### Stepper Style


![image](https://github.com/woocommerce/woocommerce/assets/4463174/b35c4356-8dd6-44a7-903d-023118dadaf4)






### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable TT4.
2. Edit the Single Product Template.
3. Ensure that you have the Add To Cart Form block added.
4. Click on it.
5. Open the block settings.
6. Ensure that you have the `Quantity Selector` option
7. Ensure that the default one is `Input`.
8. Ensure that the input looks like this image:

![image](https://github.com/woocommerce/woocommerce/assets/4463174/a2263956-e591-4dc4-ac63-33695cb6d6ee)


9. Switch the option to `Stepper`
10. Ensure that the input looks like this image:
![image](https://github.com/woocommerce/woocommerce/assets/4463174/49585883-1798-4353-afb4-62083512d278)
**The following steps should be checked for both options and with a Simple Product, a Variation Product, and a Grouped Product**
11. Visit the Product.
12. Ensure that the input looks like the image in [the table ](https://github.com/woocommerce/woocommerce/pull/47664#issuecomment-2137043597).
13. Ensure that the input works correctly (pay attention to focus/border/etc...).
14. Click Add to Cart.
15. Ensure that the right amount of the products is added.
16. Repeat with TT3 or another block theme.
17. Create a new post.
**The following steps should be checked for both options and with a Simple Product, a Variation Product and a Grouped Product**
19. Add the `Single Product` block
11. Visit the page.
12. Ensure that the input looks like the image [in the table](https://github.com/woocommerce/woocommerce/pull/47664#issuecomment-2137043597).
13. Ensure that the input works correctly (pay attention to focus/border/etc...).
14. Click Add to Cart.
15. Ensure that the right amount of the products is added.
16. Repeat with TT3 or another block theme.
17. Testing this PR with Firefox and Chrome.





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add to Cart with Options: Add Quantity Selector option.

#### Comment 

</details>
